### DISCUSSION TITLE
Version 2.0.0

### DISCUSSION THREAD

**Discussion by FelixKratz** [FelixKratz/SketchyBar#79] [Announcements]
I am currently working on completing v2.0.0 with major changes across the whole app -- Including configuration syntax.
Although I really think the reworked syntax is much better and cleaner, I dont know exactly how to make the update process as simple as possible. Because in my current local version old configurations will simply not work anymore. Does anyone have an idea how to go about deprecating parts of the old syntax without causing major headache to all those with an outdated configuration?

**Comment by es183923** [comment]
Is the new config syntax still using shell commands? If not, it should be relatively easy to detect. You could create something that converts the old config to the new one.

**Reply by FelixKratz** [reply]
Nothing has changed about the shell command nature of the configuration, the commands are a bit different (not drastically different, but different enough to brick current configs). Furthermore I have removed deprecated commands that where only supported for retro compatibility. I could also make a document that highlights all the changes and makes it easier to upgrade.

**Comment by es183923** [comment]
You could do something like: `sketchybar -m set version 2`, then, the users will be able to use the updated version. The default version could be `1` (which is what it is right now), and then at some point in the future, change to version `2`.

**Comment by FelixKratz** [comment]
I have create a document which highlights the changes in the new [release](https://github.com/FelixKratz/SketchyBar/releases/tag/v2.0.0) and gives a guide for updating.
Additionally I will create a brew formula for the old v1.4.3 (the previous HEAD version), available via
```bash
brew install sketchybarv1
```
